sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/json/JsonNodeType.java

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.json;
+
+/**
+ * The enumeration of JSON node types.
+ */
+enum JsonNodeType {
+    /**
+     * Corresponds to the 'null' value in JSON.
+     */
+    NULL,
+
+    /**
+     * Corresponds to the 'false' value in JSON.
+     */
+    FALSE,
+
+    /**
+     * Corresponds to the 'true' value in JSON.
+     */
+    TRUE,
+
+    /**
+     * Corresponds to the number type in JSON (number = [ minus ] integer [ fraction ] [ exponent ]).
+     */
+    NUMBER64,
+
+    /**
+     * Corresponds to the string type in JSON (string = quotation-mark *char quotation-mark).
+     */
+    STRING,
+
+    /**
+     * Corresponds to the array type in JSON ( begin-array [ value *( value-separator value ) ] end-array).
+     */
+    ARRAY,
+
+    /**
+     * Corresponds to the object type in JSON (begin-object [ member *( value-separator member ) ] end-object).
+     */
+    OBJECT,
+
+    /**
+     * Corresponds to the property name of a JSON object property (which is also a string).
+     */
+    FIELDNAME,
+
+    /**
+     * Corresponds to the byte type in Java for the extended types.
+     */
+    INT8,
+
+    /**
+     * Corresponds to the short type in Java for the extended types.
+     */
+    INT16,
```

Comment:
I curious should these all be different JsonNodeType or categorize them into Number

Reply:
The intention with this model is to create a 1:1 mapping between node types and return value types.  Especially in the case of the Navigator classes (ex: https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Json/JsonNavigator.JsonBinaryNavigator.cs#L526) it should simplify implementing write functions.  I would argue to leave it as is for now, and then review if it makes sense to combine these into one enum value when we revaluate aligning with the azure-core/azure-json library interfaces.

Reply:
The intention with this model is to create a 1:1 mapping between JsonNodeTypes and Java types. Especially in the case of the Navigator classes (ex:  https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Json/JsonNavigator.JsonBinaryNavigator.cs#L526) it should simplify implementing some functions. I would argue to leave it as is for now, and then review if it makes sense to combine these into one enum value when we revaluate aligning with the azure-core/azure-json library interfaces.